frame/vesting/src/lib.rs

```diff
@@ -302,121 +407,322 @@ pub mod pallet {
+	fn merge_vesting_info(
+		now: T::BlockNumber,
+		schedule1: VestingInfo<BalanceOf<T>, T::BlockNumber>,
+		schedule2: VestingInfo<BalanceOf<T>, T::BlockNumber>,
+	) -> Option<VestingInfo<BalanceOf<T>, T::BlockNumber>> {
+		let schedule1_ending_block = schedule1.ending_block::<T::BlockNumberToBalance>();
+		let schedule2_ending_block = schedule2.ending_block::<T::BlockNumberToBalance>();
+		let now_as_balance = T::BlockNumberToBalance::convert(now);
+
+		// Check if one or both schedules have ended.
+		match (schedule1_ending_block <= now_as_balance, schedule2_ending_block <= now_as_balance) {
+			// If both schedules have ended, we don't merge and exit early.
+			(true, true) => return None,
+			// If one schedule has ended, we treat the one that has not ended as the new
+			// merged schedule.
+			(true, false) => return Some(schedule2),
+			(false, true) => return Some(schedule1),
+			// If neither schedule has ended don't exit early.
+			_ => {},
+		}
+
+		let locked = schedule1
+			.locked_at::<T::BlockNumberToBalance>(now)
+			.saturating_add(schedule2.locked_at::<T::BlockNumberToBalance>(now));
+		// This shouldn't happen because we know at least one ending block is greater than now.
+		if locked.is_zero() {
+			return None;
+		}
+
+		let ending_block = schedule1_ending_block.max(schedule2_ending_block);
+		let starting_block = now.max(schedule1.starting_block()).max(schedule2.starting_block());
+		let duration =
+			ending_block.saturating_sub(T::BlockNumberToBalance::convert(starting_block));
+		let per_block = if duration.is_zero() {
+			// The logic of `ending_block` guarantees that each schedule ends at least a block
+			// after it starts and since we take the max starting and ending_block we should never
+			// get here
+			locked
+		} else if duration > locked {
+			// This would mean we have a per_block of less than 1, which should not be not possible
+			// because when we create the new schedule is at most the same duration as the longest,
+			// but never greater.
+			1u32.into()
+		} else {
+			locked.checked_div(&duration)?
+		};
+
+		// At this point inputs have been validated, so this should always be `Some`.
+		let schedule = VestingInfo::new::<T>(locked, per_block, starting_block);
+		debug_assert!(schedule.validate::<T>().is_ok());
+		Some(schedule)
+	}
+
+	// Execute a vested transfer from `source` to `target` with the given `schedule`.
+	fn do_vested_transfer(
+		source: <T::Lookup as StaticLookup>::Source,
+		target: <T::Lookup as StaticLookup>::Source,
+		schedule: VestingInfo<BalanceOf<T>, T::BlockNumber>,
+	) -> DispatchResult {
+		ensure!(schedule.locked() > T::MinVestedTransfer::get(), Error::<T>::AmountLow);
+		let schedule = schedule.validate::<T>()?;
+
+		let target = T::Lookup::lookup(target)?;
+		let source = T::Lookup::lookup(source)?;
+		if let Some(len) = Vesting::<T>::decode_len(&target) {
+			ensure!(
+				len < T::MaxVestingSchedules::get() as usize,
+				Error::<T>::AtMaxVestingSchedules
+			);
+		}
```

Comment:
```suggestion
		ensure!(Vesting::<T>::decode_len(&target).unwrap_or_default() < T::MaxVestingSchedules::get() as usize);
```

I belive they are the same